Mathlib/CategoryTheory/Localization/Monoidal/Braided.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2025 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Localization.Monoidal.Basic
+import Mathlib.CategoryTheory.Monoidal.Braided.Basic
+
+/-!
+
+# Localization of symmetric monoidal categories
+
+Let `C` be a monoidal category equipped with a class of morphisms `W` which
+is compatible with the monoidal category structure. The file
+`Mathlib.CategoryTheory.Localization.Monoidal.Basic` constructs a monoidal structure on
+the localized on `D` such that the localization functor is monoidal.
+
+In this file we promote this monoidal structure to a braided structure in the case where `C` is
+braided, in such a way that the localization functor is braided. If `C` is symmetric monoidal, then
+the monoidal structure on `D` is also symmetric.
+-/
+
+open CategoryTheory Category MonoidalCategory BraidedCategory
+
+namespace CategoryTheory.Localization.Monoidal
+
+variable {C D : Type*} [Category C] [Category D] (L : C ⥤ D) (W : MorphismProperty C)
+  [MonoidalCategory C] [W.IsMonoidal] [L.IsLocalization W]
+  {unit : D} (ε : L.obj (𝟙_ C) ≅ unit)
+
+local notation "L'" => toMonoidalCategory L W ε
+
+instance : (L').IsLocalization W := inferInstanceAs (L.IsLocalization W)
+
+section Braided
+
+variable [BraidedCategory C]
+
+noncomputable instance : Lifting₂ L' L' W W ((curriedTensor C).flip ⋙ (whiskeringRight C C
+    (LocalizedMonoidal L W ε)).obj L') (tensorBifunctor L W ε).flip :=
+  inferInstanceAs (Lifting₂ L' L' W W (((curriedTensor C) ⋙ (whiskeringRight C C
+    (LocalizedMonoidal L W ε)).obj L')).flip (tensorBifunctor L W ε).flip )
+
+/-- The braiding on the localized category as a natural isomorphism of bifunctors. -/
+noncomputable def braidingNatIso : tensorBifunctor L W ε ≅ (tensorBifunctor L W ε).flip :=
+  lift₂NatIso L' L' W W
+    ((curriedTensor C) ⋙ (whiskeringRight C C
+      (LocalizedMonoidal L W ε)).obj L')
+    (((curriedTensor C).flip ⋙ (whiskeringRight C C
+      (LocalizedMonoidal L W ε)).obj L'))
+    _ _  (isoWhiskerRight (NatIso.ofComponents (fun X ↦ NatIso.ofComponents (fun Y ↦ β_ X Y))) _)
```

Comment:
Similarly as we have `curriedAssociatorNatIso` in `Monoidal.Basic`, I think we should add `curriedBraidingNatIso` (from `curriedTensor C` to its flip) in the file `Monoidal.Braided.Basic`. (It could be defined just after `tensorLeftIsoTensorRight`.) Then, this definition could probably be used here.